samples/snippets/src/main/java/com/example/analytics/RunReportWithDateRangesSample.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.analytics;
+
+/* Google Analytics Data API sample application demonstrating the usage of
+date ranges in a report.
+
+See
+https://developers.google.com/analytics/devguides/reporting/data/v1/rest/v1beta/properties/runReport#body.request_body.FIELDS.date_ranges
+for more information.
+
+Before you start the application, please review the comments starting with
+"TODO(developer)" and update the code to use correct values.
+
+To run this sample using Maven:
+  cd java-analytics-data/samples/snippets
+  mvn compile
+  mvn exec:java -Dexec.mainClass="com.example.analytics.RunReportWithDateRangesSample"
```

Comment:
Can we use exec:exec instead of exec:java?

Reply:
Hey Averi. Just chiming in on behalf of the Google Analytics team. Is it possible you can point us to any instance of other samples using exec:exec? It does not seem to be possible to run just a single class using exec:exec, as it does not support -Dexec.mainClass argument. https://github.com/GoogleCloudPlatform/java-docs-samples/pull/1523 just hardcodes the classname in pom.xml which works with a single Quickstart sample, but we have dozens of them. Thanks!

Reply:
I see that guidance doesn't work well here. You can continue to use exec:java